mne/datasets/_fetch.py

```diff
@@ -42,7 +75,7 @@ def fetch_dataset(
     accept=False,
     auth=None,
     token=None,
-):
+) -> Union[Path, tuple[Path, str]]:
```

Comment:
@cbrnr `Union` is used here, but I'm inferring that it isn't enough by itself

Reply:
Yes, the problem is that the return type depends on the argument, so I guess there's no way around `@overload`

Reply:
Annotating only the `Union` and omitting the overloads makes it impossible for the type checker to infer which of the two possible return types (in this specific case, Path or tuple of Path and string) to expect based on the parameters provided to the function